test/testtokenize.cpp

```diff
@@ -8810,6 +8814,30 @@ class TestTokenizer : public TestFixture {
         SimpleTokenizer tokenizer(settingsDefault, *this, false);
         ASSERT_NO_THROW(tokenizer.tokenize(code));
     }
+
+    void simplifyEnum1() {
+        const char code[] = "static enum {A,B} ab;";
+        ASSERT_EQUALS("enum Anonymous0 { A , B } ; static enum Anonymous0 ab ;", tokenizeAndStringify(code));
+        SimpleTokenizer tokenizer(settingsDefault, *this);
+        tokenizer.tokenize(code);
+        const Token* tok = Token::findsimplematch(tokenizer.tokens(), "static");
+        ASSERT(tok);
+        ASSERT_EQUALS(tok->column(), 1);
+        ASSERT_EQUALS(tok->next()->column(), 8);
+        ASSERT_EQUALS(tok->tokAt(2)->column(), 0);
+    }
+
+    void simplifyEnum2() {
+        const char code[] = "enum static E {A,B} ab;";
+        ASSERT_EQUALS("enum E { A , B } ; static enum E ab ;", tokenizeAndStringify(code));
+        SimpleTokenizer tokenizer(settingsDefault, *this);
+        tokenizer.tokenize(code);
+        const Token* tok = Token::findsimplematch(tokenizer.tokens(), "static");
+        ASSERT(tok);
+        ASSERT_EQUALS(tok->column(), 0);
```

Comment:
I would suggest that we try to write a proper column value here. It would be shaky for premiumaddon to warn about `static` whenever column is 0 => since it is not known why it is 0.